statchatui/src/Compare/Compare.jsx

```diff
@@ -21,10 +21,53 @@ const Compare = ({ signedInUserData }) => {
   const [isLoading, setIsLoading] = useState(false);
   const [otherUserStats, setOtherUserStats] = useState(null);
   const [signedInUserStats, setSignedInUserStats] = useState(null);
+  const [userQueue, setUserQueue] = useState([]);
+  const [showStats, setShowStats] = useState(null);
+  const storedUserQueue = sessionStorage.getItem("userQueue");
+
+  //get the previously stored userqueue everytime the page mounts
+  useEffect(() => {
+    if (storedUserQueue) {
+      setUserQueue(JSON.parse(storedUserQueue));
+    }
+  }, []);
+
+  //everytime userqueue is updated, sessionStorage will be updated with the new value.
+  useEffect(() => {
+    sessionStorage.setItem("userQueue", JSON.stringify(userQueue));
+  }, [userQueue]);
+
+  const compareUsers = async (user1, user2) => {
+    setIsLoading(true); //sets the state to true before the api call is displaying the stats
+    const stats1 = await fetchStats(user1);
+    const stats2 = await fetchStats(user2);
+
+    setOtherUserStats(stats1);
+    setSignedInUserStats(stats2);
+    setOtherUserFormData(user1);
+    setSignedInUserFormData(user2);
+
+    const newUser = {
+      gamertag: user1.gamertag,
+      platform: user1.platform,
+      gameName: user1.gameName,
+      // Assigned the newUser to have the stats of stats1 because otherUser and newUser are the same and I want the stats to be saved in the userQueue
+      stats: stats1,
+    };
+
+    setUserQueue((prevQueue) => {
+      const updatedQueue = [...prevQueue, newUser];
+      // if the queue size is over three, remove oldest user which is the first element of the array
+      if (updatedQueue.length > 3) {
+        updatedQueue.shift();
+      }
```

Comment:
Here's a few test cases for you! Do you currently handle these cases?
1. User searches: "alice", "bob", "alice"
2. User searches: "alice", "alice", "alice"
3. User searches: "alice", "sdlfkjsdlfjksefsdkf" (some invalid username), "bob"
4. User searches: "alice", "", "bob", "caleb"

Reply:
i will handle these in the error PR, i will only add valid users to the user queue

Reply:
also i will not allow repeat users

Reply:
That sounds good to me, but it would be helpful to mention that there will be a second PR to improve the algorithm in your description